py-polars/polars/lazyframe/frame.py

```diff
@@ -622,6 +622,26 @@ def schema(self) -> SchemaDict:
         """
         return self._ldf.schema()
 
+    def __dataframe_consortium_standard__(
+        self, /, *, api_version: str | None = None
+    ) -> Any:
+        """
+        Provide entry point to the Consortium DataFrame Standard API.
+
+        This is developed and maintained outside of polars.
+        Please report any issues to https://github.com/data-apis/dataframe-api-compat.
+        """
+        try:
+            from dataframe_api_compat import polars_standard  # type: ignore[import]
+        except ModuleNotFoundError:
+            raise ImportError(
+                "`dataframe-api-compat` package is required for using the "
+                "Consortium DataFrame Standard API."
+            ) from None
+        return polars_standard.convert_to_standard_compliant_dataframe(
+            self, api_version=api_version
+        )
```

Comment:
Does this function check for `LazyFrame`? Or how does this work exactly?

If it collects behind the scenes, I think we should not allow this method on LazyFrames and only have it on DataFrames.

Reply:
yup, the aim is that it _should_ all work for lazyframes - if you start with a lazyframe, the operation will return a lazyframe

In practice, the implementation isn't there yet, but that can be handled there (and libraries can then just set a minimum version on `dataframe-api-compat`, which should be easier than waiting for everything to be 100% and having to set a very high minimum version on polars itself)